core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -388,7 +388,7 @@ private static String toSql(RelNode root, SqlDialect dialect,
         + "where \"net_weight\" <> 10 or \"net_weight\" is null";
     final String expected = "SELECT \"product_id\", \"shelf_width\"\n"
         + "FROM \"foodmart\".\"product\"\n"
-        + "WHERE \"net_weight\" <> 10 OR \"net_weight\" IS NULL";
+        + "WHERE \"net_weight\" <> CAST(10 AS DOUBLE) OR \"net_weight\" IS NULL";
```

Comment:
Would users need to make changes to queries such as these? Is 10 no longer implicitly converted to double when comparing to double columns?

Reply:
no, this is just the canonical representation of the program created by Calcite which proves that the 10 is implicitly cast to a double